Allow help2man to run qmicli and qmi-firmware-update through an exe wrapper (qemu-user) when cross-compiling
--- a/docs/man/meson.build
+++ b/docs/man/meson.build
@@ -3,13 +3,27 @@
 
 help2man = find_program('help2man')
 
+# help2man runs the program through the shell, so a wrapper (e.g. qemu-user)
+# can be prepended to run built programs when cross-compiling
+man_exe_wrapper = get_option('man_exe_wrapper')
+
+qmicli_man = qmicli
+if man_exe_wrapper != ''
+  qmicli_man = man_exe_wrapper + ' ' + qmicli.full_path()
+endif
+
 mans = [
-  ['qmicli', [help2man, '--output=@OUTPUT@', '--name=Control QMI devices', '--help-option="--help-all"', qmicli]],
-  ['qmi-network', [help2man, '--output=@OUTPUT@', '--name=Simple network management of QMI devices', qmi_network]],
+  ['qmicli', [help2man, '--output=@OUTPUT@', '--name=Control QMI devices', '--help-option="--help-all"', qmicli_man], [qmicli]],
+  ['qmi-network', [help2man, '--output=@OUTPUT@', '--name=Simple network management of QMI devices', qmi_network], []],
 ]
 
 if enable_firmware_update
-  mans += [['qmi-firmware-update', [help2man, '--output=@OUTPUT@', '--name=Update firmware in QMI devices', qmi_firmware_update]]]
+  qmi_firmware_update_man = qmi_firmware_update
+  if man_exe_wrapper != ''
+    qmi_firmware_update_man = man_exe_wrapper + ' ' + qmi_firmware_update.full_path()
+  endif
+
+  mans += [['qmi-firmware-update', [help2man, '--output=@OUTPUT@', '--name=Update firmware in QMI devices', qmi_firmware_update_man], [qmi_firmware_update]]]
 endif
 
 foreach man: mans
@@ -17,6 +31,7 @@
     man[0],
     output: man[0] + '.1',
     command: man[1],
+    depends: man[2],
     install: true,
     install_dir: qmi_mandir / 'man1',
   )
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -17,6 +17,7 @@
 option('introspection', type: 'boolean', value: true, description: 'build introspection support')
 option('gtk_doc', type: 'boolean', value: false, description: 'use gtk-doc to build documentation')
 option('man', type: 'boolean', value: true, description: 'build man pages using help2man')
+option('man_exe_wrapper', type: 'string', value: '', description: 'wrapper used by help2man to run built programs (e.g. when cross-compiling)')
 
 option('bash_completion', type: 'boolean', value: true, description: 'install bash completion files')
 
